### PR TITLE
Only match with viper config if it was read

### PR DESCRIPTION
**Pull request by vkamra**
Fix a regression introduced in 1ba103523ae. The `matchFromConfig` block should only be entered if we read config from viper.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3856 

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/cli/config/account.go [hunk 1]
```diff
@@ -50,16 +50,16 @@ func configureAccount(
...
 
-	if matchFromConfig {
-		providerType := vpr.GetString(AccountProviderTypeKey)
-		if providerType != account.ProviderM365.String() {
-			return acct, clues.New("unsupported account provider: " + providerType)
-		}
+		if matchFromConfig {
+			providerType := vpr.GetString(AccountProviderTypeKey)
+			if providerType != account.ProviderM365.String() {
+				return acct, clues.New("unsupported account provider: " + providerType)
```
Comment (pandeyabs) [inline src/cli/config/account.go:57]:
```suggestion
				return acct, clues.New("unsupported account provider " + providerType)
```

File: src/cli/config/storage.go [hunk 2]
```diff
@@ -75,16 +75,16 @@ func configureStorage(
...
 
-	if matchFromConfig {
-		providerType := vpr.GetString(StorageProviderTypeKey)
-		if providerType != storage.ProviderS3.String() {
-			return store, clues.New("unsupported storage provider: " + providerType)
-		}
+		if matchFromConfig {
+			providerType := vpr.GetString(StorageProviderTypeKey)
+			if providerType != storage.ProviderS3.String() {
+				return store, clues.New("unsupported storage provider: " + providerType)
```
Comment (pandeyabs) [inline src/cli/config/storage.go:82]:
```suggestion
				return store, clues.New("unsupported storage provider " + providerType)
```

**Review by pandeyabs — COMMENTED** [review]
Changes look good to me! Since I am not super familiar with config handling, we should wait for @neha-Gupta1 to review it as well.

**Comment by pandeyabs** [comment]
Update: Removed my suggested nits, weren't necessary.

**Review by neha-Gupta1 — APPROVED** [review]
The changes do make sense. Because anyhow if config file could not be read matching values won't work. 
Approving to fix the release bug.

**Pull request merged**